packages/protocol/test/unit/balanceSheet/view/getRepayBondAmount.ts

```diff
@@ -0,0 +1,89 @@
+import { expect } from "chai";
+import { getRepayBondAmount } from "../../../shared/mirrors";
+
+export default function shouldBehaveLikeGetRepayBondAmount(): void {
+  context("when the liquidation incentive is 100%", function () {
+    beforeEach(async function () {
+      await this.mocks.fintroller.mock.getLiquidationIncentive
+        .withArgs(this.mocks.wbtc.address)
+        .returns(LIQUIDATION_INCENTIVES.lowerBound);
+    });
+
+    it("returns zero", async function () {
+      const liquidationAmount: BigNumber = hUSDC("15000");
+      const repayAmount: BigNumber = await this.contracts.balanceSheet.getRepayBondAmount(
+        this.mocks.hTokens[0].address,
+        liquidationAmount,
+        this.mocks.wbtc.address,
+      );
+      expect(repayAmount).to.equal(Zero);
+    });
+  });
+
+  context("when the liquidation incentive is not 100%", function () {
+    beforeEach(async function () {
+      await this.mocks.fintroller.mock.getLiquidationIncentive
+        .withArgs(this.mocks.wbtc.address)
+        .returns(LIQUIDATION_INCENTIVES.default);
+    });
+
+    context("when the collateral liquidation amount is zero", function () {
+      it("returns zero", async function () {
+        const liquidationAmount: BigNumber = hUSDC("0");
+        const repayAmount: BigNumber = await this.contracts.balanceSheet.getRepayBondAmount(
+          this.mocks.hTokens[0].address,
+          liquidationAmount,
+          this.mocks.wbtc.address,
+        );
+        expect(repayAmount).to.equal(Zero);
+      });
+    });
+
+    context("when the collateral liquidation amount is not zero", function () {
+      const liquidationAmount: BigNumber = hUSDC("15000");
+
+      context("when the collateral has 18 decimals", function () {
+        const collateralDecimals: BigNumber = bn("18");
+
+        beforeEach(async function () {
+          await this.mocks.wbtc.mock.decimals.returns(collateralDecimals);
+        });
+
+        it("retrieves the correct value", async function () {
+          const repayAmount: BigNumber = await this.contracts.balanceSheet.getRepayBondAmount(
+            this.mocks.hTokens[0].address,
+            liquidationAmount,
+            this.mocks.wbtc.address,
+          );
+          expect(repayAmount).to.equal(
+            getRepayBondAmount(liquidationAmount, NORMALIZED_WBTC_PRICE, collateralDecimals),
+          );
+        });
+      });
+
+      context("when the collateral has 8 decimals", function () {
+        const collateralDecimals: BigNumber = bn("8");
+
+        beforeEach(async function () {
+          await this.mocks.wbtc.mock.decimals.returns(collateralDecimals);
+        });
+
+        it("retrieves the correct value", async function () {
+          const repayAmount: BigNumber = await this.contracts.balanceSheet.getRepayBondAmount(
+            this.mocks.hTokens[0].address,
+            liquidationAmount,
+            this.mocks.wbtc.address,
+          );
+          expect(repayAmount).to.equal(
```

Comment:
@paulrberg I added the unit tests for `getRepayBondAmount(...)`, but this particular line is failing when the number of decimals is 8. All other cases work just fine. I'm guessing this is due to bad math conversion either in the Solidity file or in the `mirrors.ts` function I added for the expected output, would be great if you could take a look.

Reply:
There were two issues:

1. You were using the `hUSDC` helper function to create a collateral amount.
2. You were not de-normalising the collateral amount (created mistakenly with `hUSDC`).

Reply:
*Update*: I am still getting an error after correcting the errors above. There must be a third error.

Reply:
Yup, the third error was in the Solidity code. You were using the PRBMath function `mul` on two values which were not normalised to 18 decimals. Only normalised amounts can be passed to PRBMath. You should have used the normal `*` operator.

Side note: this wouldn't have happened if we used the [typed version of UD60x18](https://github.com/hifi-finance/prb-math/blob/061ab63c61732b3e9ad037767be82874aec63094/contracts/PRBMathUD60x18Typed.sol). I am considering to switch over to that .. although this sort of bugs should always be caught by tests.

Reply:
Ok, so I've replaced `hUSDC` with `WBTC` helper function, replaced the `.mul` with the default Solidity multiplication, and added the unchecked math block.

Reply:
Check Slack.

Reply:
I fixed all of these on my local branch, plus more. I will have to git push with `--force`.